source/numeric-range.d.ts

```diff
@@ -0,0 +1,46 @@
+/**
+A `number` that is an integer.
+
+Use-case: Validating and documenting parameters.
+
+@example
+```
+declare function setYear<T extends number>(length: Integer<T>): void;
+```
+*/
+export type Integer<T extends number> = `${T}` extends `${bigint}` ? T : never;
```

Comment:
This is really cool! I have been wanting in integer type for a long time. It would be nice with a short code comment explaining how this works.

Reply:
Maybe add a caveat to the docs that it will not detect overflowing integer literal.

Reply:
I updated the types to make `Integer` return `never` when given [positive or negative infinity as a type](https://github.com/microsoft/TypeScript/issues/32277).